src/extension/method/provider/provider.ts

```diff
@@ -1,6 +1,6 @@
 import { functionMethod } from './functionMethod';
 // tslint:disable-next-line:no-any
-type Method = (name: string, value: any) => () => any;
+type Method = (name: string, value: () => any) => () => any;
```

Comment:
This is a breaking change isn't it? Do we want to go through a deprecation phase before or do you think it's ok going forward with this?

Deprecation would be like:
• adding a new type MethodWithDeferredValue
• adding a new method provideMethodWithDeferredValue
• deprecate provideMethod
• change provideMethod wrapping the provided method and calling value() before passing it (to match the old behaviour)

Let me know

Reply:
I think it make sense! Its also a good exercise for us to finally have a library that doesnt do breaking changes immediately and we dont have to move to v2 until we are ready. I have to document the deprecation and explain in which scenario the compiler will die with max call stack

Reply:
done